src/lua/datetime.lua

```diff
@@ -0,0 +1,623 @@
+local ffi = require('ffi')
+
+--[[
+    `c-dt` library functions handles properly both positive and negative `dt`
+    values, where `dt` is a number of dates since Rata Die date (0001-01-01).
+
+    For better compactness of our typical data in MessagePack stream we shift
+    root of our time to the Unix Epoch date (1970-01-01), thus our 0 is
+    actually dt = 719163.
+
+    So here is a simple formula how convert our epoch-based seconds to dt values
+        dt = (secs / 86400) + 719163
+    Where 719163 is an offset of Unix Epoch (1970-01-01) since Rata Die
+    (0001-01-01) in dates.
+]]
+
+ffi.cdef [[
+
+/* dt_core.h definitions */
+typedef int dt_t;
+
+typedef enum {
+    DT_MON       = 1,
+    DT_MONDAY    = 1,
+    DT_TUE       = 2,
+    DT_TUESDAY   = 2,
+    DT_WED       = 3,
+    DT_WEDNESDAY = 3,
+    DT_THU       = 4,
+    DT_THURSDAY  = 4,
+    DT_FRI       = 5,
+    DT_FRIDAY    = 5,
+    DT_SAT       = 6,
+    DT_SATURDAY  = 6,
+    DT_SUN       = 7,
+    DT_SUNDAY    = 7,
+} dt_dow_t;
+
+dt_t   tnt_dt_from_rdn     (int n);
+dt_t   tnt_dt_from_ymd     (int y, int m, int d);
+void   tnt_dt_to_ymd       (dt_t dt, int *y, int *m, int *d);
+void   tnt_dt_to_yqd       (dt_t dt, int *y, int *q, int *d);
+void   tnt_dt_to_ywd       (dt_t dt, int *y, int *w, int *d);
+
+int    tnt_dt_rdn          (dt_t dt);
+dt_dow_t tnt_dt_dow        (dt_t dt);
+
+/* dt_util.h */
+bool    tnt_dt_leap_year       (int y);
+int     tnt_dt_days_in_year    (int y);
+int     tnt_dt_days_in_quarter (int y, int q);
+int     tnt_dt_days_in_month   (int y, int m);
+int     tnt_dt_weeks_in_year   (int y);
+
+/* dt_accessor.h */
+
+int     tnt_dt_year         (dt_t dt);
+int     tnt_dt_month        (dt_t dt);
+int     tnt_dt_doy          (dt_t dt);
+int     tnt_dt_dom          (dt_t dt);
+
+/* dt_arithmetic.h definitions */
+
+typedef enum {
+    DT_EXCESS,
+    DT_LIMIT,
+    DT_SNAP
+} dt_adjust_t;
+
+dt_t   tnt_dt_add_years    (dt_t dt, int delta, dt_adjust_t adjust);
+dt_t   tnt_dt_add_quarters (dt_t dt, int delta, dt_adjust_t adjust);
+dt_t   tnt_dt_add_months   (dt_t dt, int delta, dt_adjust_t adjust);
+
+/* dt_parse_iso.h definitions */
+
+size_t tnt_dt_parse_iso_date (const char *str, size_t len, dt_t *dt);
+size_t tnt_dt_parse_iso_time (const char *str, size_t len, int *sod, int *nsec);
+size_t tnt_dt_parse_iso_zone_lenient(const char *str, size_t len, int *offset);
+
+/* Tarantool functions - datetime.c */
+
+int    datetime_to_string(char *buf, int len, const struct datetime * date);
+size_t datetime_strftime(char *buf, uint32_t len, const char *fmt,
+                        const struct datetime *date);
```

Comment:
You are using struct datetime without its announcement in this cdef.

Reply:
Because it's being defined by `tarantool_lua_utils_init` in `utils.c`